plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -288,6 +288,12 @@ public Optional<JdbcTableHandle> getTableHandle(ConnectorSession session, Schema
         return get(tableHandleCache, key, () -> delegate.getTableHandle(session, schemaTableName));
     }
 
+    @Override
+    public JdbcTableHandle getTableHandle(ConnectorSession session, PreparedQuery preparedQuery)
+    {
+        return delegate.getTableHandle(session, preparedQuery); // TODO add caching?
```

Comment:
I think we should. @kokosing thoughts?

(note: `PreparedQuery` defines value-based equals, so ok from this perspective)

Reply:
Looks ok to cache it.